js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,72 @@
+describe("enhancedDate Test", function () {
+
+    describe("Testing setDate method", function () {
+        it("should be an instance of a Date object", function () {
+            expect(enhancedDate.setDate()).is.an.instanceof(Date);
+        });
+        it("should equal to now", function () {
+            expect(enhancedDate.setDate().getTime()).to.deep.equal(Date.now());
+        });        
+    });
+    
+    describe("Testing getDate method", function() {
+        it("should have type number", function () {
+            expect(enhancedDate.getDate()).to.be.a("number");
+        });
+        it("should be close to NOW in milliseconds", function () {
+            expect(enhancedDate.getDate()).to.be.closeTo(Date.now(), 10);
+        });
+        it("should be an instance of Date object", function () {
+            expect(enhancedDate.getDate(new Date())).is.an.instanceof(Date);
```

Comment:
getDate when it takes a value, is just checking whether it's a "truthy" value, so while passing in a Date instance is "truthy", the test would be more clear if it was just "true".